agent/agent.go

```diff
@@ -1270,8 +1267,17 @@ func (a *agent) startReportingConnectionStats(ctx context.Context) {
 		// Convert from microseconds to milliseconds.
 		stats.ConnectionMedianLatencyMS /= 1000
 
+		lastStat := a.latestStat.Load()
+		if lastStat != nil && reflect.DeepEqual(lastStat, stats) {
+			a.logger.Info(ctx, "skipping stat because nothing changed")
+			return
+		}
+
 		select {
 		case a.connStatsChan <- stats:
+			// Only store the latest stat when it's successfully sent!
+			// Otherwise, it should be sent again on the next iteration.
+			a.latestStat.Store(stats)
```

Comment:
Considering the previous comment about Tailscale resetting counts on every report, I'd think this current implementation will lose stats? 

I imagine a more safe way to update the stats would be something along the lines of:

```go
a.statMu.Lock()
a.stats.RxBytes += ...

select {
case a.connStatsChan <- a.stats: // note: a copy assuming basic struct
	// sent
default:
	// dropped this report
}
a.statMu.Unlock()

This way we're always incrementing the numbers, even from dropped reports.

Reply:
Ahh, good point!

Reply:
I wonder if we'd be better off blocking instead of dropping? It seems like that's fine from the Tailscale side, and then we don't really run any risks here.

Reply:
I changed it to block instead. Let me know your thoughts! The loop retries anyways.